extensions/amp-story/1.0/bookend/components/cta-link.js

```diff
@@ -0,0 +1,96 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {BookendComponentInterface} from './bookend-component-interface';
+import {addAttributesToElement} from '../../../../../src/dom';
+import {dict} from '../../../../../src/utils/object';
+import {htmlFor} from '../../../../../src/static-template';
+import {isProtocolValid} from '../../../../../src/url';
+import {user} from '../../../../../src/log';
+
+/**
+ * @typedef {{
+ *   type: string,
+ *   text1: string,
+ *   url1: string,
+ *   text2: string,
+ *   url2: string
+ * }}
+ */
+export let CtaLinkDef;
+
+const TAG = 'amp-story-bookend';
+
+/**
+ * Builder class for the call to action button pair.
+ * @implements {BookendComponentInterface}
+ */
+export class CtaLinkComponent {
+  /** @override */
+  assertValidity(ctaLinksJson) {
+    if (!ctaLinksJson['links'] && !ctaLinksJson['links'].length > 0) {
```

Comment:
As a side note, this is where JavaScript gets really dangerous, since this is actually still valid, but basically the opposite of what you want:

```javascript
!["a", "b", "c"].length > 0
!3 > 0
false > 0
false
```

or

```javascript
![].length > 0
!0 > 0
true > 0
true
```